pkgs/applications/video/kodi-packages/elementum/default.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv, lib, pkgs, autoPatchelfHook, buildKodiAddon, ... }:
+let
+  addonDir = "/share/kodi/addons";
+  python = pkgs.kodi.pythonPackages.python.withPackages (p: with p; [ flake8 ]);
+  os = "${stdenv.hostPlatform.system}";
+  osname = {
+    x86_64-linux = "linux_x64";
+    x86_64-darwin = "darwin_x64";
+    i686-linux = "linux_x86";
+    armv7l-linux = "linux_armv7";
+    aarch64-linux = "linux_arm64";
+  }."${os}" or (throw "Unsupported system: ${os}");
+  hash = {
+    x86_64-linux = "sha256-lmaDBKX0zkMVHnKyH6RPBd1TKINoO3guD15v7OoX094=";
+    x86_64-darwin = "sha256-yw3HZdWB+Cm+uUhURWnM5PVKBajAcM/9EiAX+q/aE2Q=";
+    i686-linux = "sha256-D9ahCGwOmPi6hCD1k98pN1EyaiX7msgGnA8sc6VAcmo=";
+    armv7l-linux = "sha256-FIM1YmVg/S65HZOD4qZp3Qv21YsmALMu5jpH3w1x8h0=";
+    aarch64-linux = "sha256-heIjCEjADD9Jj2igcXWO9rizopql6b2d8ycdAo/cQ4k=";
+  }."${os}" or (throw "Unsupported system: ${os}");
+
+in
+buildKodiAddon rec {
+  pname = "elementum";
+  namespace = "plugin.video.elementum";
+  version = "0.1.83";
+
+  src = pkgs.fetchurl {
+    url = "https://github.com/elgatito/plugin.video.elementum/releases/download/v${version}/plugin.video.elementum-${version}.${osname}.zip";
+    sha256 = hash;
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    python
+    pkgs.unzip
+  ];
+  buildInputs = with pkgs; [
+    stdenv.cc.cc.lib
+  ];
+
+  patchPhase = ''
+    rm Makefile
+  '';
+
+  buildPHase = ":";
+
+  installPhase = ''
```

Comment:
I think you can make the built-in `installPhase` work 🤔

https://github.com/NixOS/nixpkgs/blob/9bc0b2f7912fcc2765caffbdb0cfd8242a9e4cfc/pkgs/applications/video/kodi/build-kodi-addon.nix#L13

Reply:
i'm probably dealing with sources in zip file wrong, kodi addon general installPhase tries to cd into the sources(the zip file instead of unpacked dir)
i could probably set correct $src/$sourceDir in preInstallPhase or something
Which way is a better one?

```
@nix { "action": "setPhase", "phase": "installPhase" }
installing
/nix/store/r28nip42zvcr35lsndbgafqa9w4d91n0-stdenv-linux/setup: line 1347: cd: /nix/store/ncilfz2l1a61v10k9bfa3vb3xjcg6i65-plugin.video.elementum-0.1.83.linux_x64.zip/: Not a directory
```